### PR TITLE
Add build and test jobs for operator repo

### PR DESCRIPTION
**Pull request by sdake**

### CODE REVIEW

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (icygalz) [inline prow/cluster/jobs/istio/operator/istio.operator.master.yaml]: is this supposed to be prow/test.sh ?

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (icygalz) [inline prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: which node pool do you want to use for this? build-pool or test-pool?

Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: Not sure what the delta is?

Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: Not sure what the delta is?

Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: Not sure what the delta is?

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml]: Thanks will resolve when back in the office.

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (icygalz) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: you didn't specify:
nodeSelector:
        testing: test-pool

Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: what I mean is what is the difference between the build-pool and test-pool?

Cheers
-steve

Comment (utka) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: At the moment test-pool has more nodes 20 vs build-pool 10.

Comment (sdake) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: is the prow rebase on top of master prow done?

Cheers
-steve

**Comment by sdake** [comment]
Depends on this PR: https://github.com/istio/operator/pull/62

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (utka) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: work in progress

Comment (fejta) [reply prow/cluster/jobs/istio/operator/istio.operator.master.yaml:50]: yes, done

**Comment by sdake** [comment]
this PR sould be ready to go, unless there has been some alteration in the prow configuration that prevents it from working.

Cheers
-steve

**Review by fejta — COMMENTED** [review]
Please create an OWNERS file in https://github.com/istio/test-infra/tree/9c70ade9e5481f6c2ac128119430dd37d18e591e/prow/cluster/jobs/istio/operator

These reviews should be self-service for the operator folks. There is no need to gate things on the test-infra and/or oncall team.

**Comment by sdake** [comment]
@fejta will do although all or most of the repos are going to a model based upon codeowners rather than the owners file and MERGIFY to merge changes into the repo instead of tide.  Just thought I’d add a little context on where the project overall is heading.  I expect these changes may be coming to test infra repo as well but I am uncertain.

In the meantime I will add both files and we can sort out cleanup in the future.

Good idea on self service.

Cheers
Steve

**Comment by icygalz** [comment]
FYI, we want to separate the nodes for building images and running tests
whenever possible(maybe for security reasons?).
Build-pool: only for building images. Not running any tests.
Test-pool: for (build and) running tests.

File: prow/cluster/jobs/istio/operator/CODEOWNERS
Comment (howardjohn) [inline prow/cluster/jobs/istio/operator/CODEOWNERS]: I don't think this works in subdirectories https://help.github.com/en/articles/about-code-owners#codeowners-file-location

I don't think we should even use this -- if we are going to switch to codeowners then we can do it all in one PR, no need to make it more complicated now

File: prow/cluster/jobs/istio/operator/istio.operator.master.yaml
Comment (fejta) [inline prow/cluster/jobs/istio/operator/istio.operator.master.yaml:40]: FYI: I recommend writing plain yaml configs (generating via additional tooling when necessary) and avoiding the use of complicated yaml functionality like anchors -- this complexity adds friction that decreases the number of people willing to modify jobs.

**Review by fejta — APPROVED** [review]
Looks like Diana's concerns are addressed?
/lgtm
/hold

Diana can you `/hold cancel` if you're happy? Steve feel free to do this as well if need too.

**Comment by fejta** [comment]
/hold cancel

Let's merge and iterate if something is wrong?

**Review by utka — APPROVED** [review]

**Pull request merged**
